Configure repo for use with GitLab CI/CD.

jpetto opened the pull request.

jpetto committed:
Configure repo for use with GitLab CI/CD.

pmac commented on bin/build.sh:
```diff
@@ -1,20 +1,39 @@
+# branch names don't correspond 1:1 with config file names
+branch="$1"
+case branch in
+    --dev)
```
you're looking for `--dev` on the CLI, but in the gitlab YAML you're just passing the branch name. I think here you only want e.g. `dev)` and `master)` without the leading `--`.

pmac commented on bin/build.sh:
```diff
@@ -1,20 +1,39 @@
-    --entrypoint /bin/bash \
-    huli/gulp \
-    -c "/entrypoint.sh build --production && chown $(id -u):$(id -g) . -R"
+if [[ -n $SITE ]]; then
```
NIT: best to put double quotes around the variable like `"$SITE"` to ensure that it's interpreted as an empty string if it is, otherwise bash could consider it an error because nothing was passed to `-n`.

pmac replied:
Also, if you only want to build the site for these branches, you could instead just add this too the YAML for the build job:

```yaml
only:
  - dev
  - master
  - prod
```

But see my comment on the separation of jobs themselves on why we probably don't want this either.

pmac commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,34 @@
+  - build
+  - push
+
+build site:
```
having the build as a separate "job" here means that it's possible that a different runner could get the push job than the build and that would not work at all I think. I believe for this you should just move this script to the push jobs:

```yaml
push dev:
  tags:
    - meao
  stage: push
  only: dev
  script:
    - bin/build.sh dev
    - bin/push-gh-pages.sh
```

pmac reviewed (changes requested):
I've made some comments. Let's discuss.

jpetto committed:
Review fixes:

- re-structure gitlab yaml to have one job per env and remove stages
- remove complexity from build.sh

jpetto commented:
@pmac Updated based on our convo today. Think I've got all things covered.

Also, Slack notifications are configured in the GitLab repo, but not yet marked as active.

Even if all code here is approved, still waiting on AWS profile name to be verified (created?) on the GitLab Runner box. (Checking with Josh on that tomorrow.)

bookshelfdave commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,23 @@
+  only: master
+  script:
+    - bin/build.sh stage
+    - "cd release && aws s3 sync . s3://awebpodcast-stage --acl public-read --delete --profile awebpodcast"
```
we don't need the `--profile awebpodcast` anymore now that we're using an IAM role on the gitlab runner.

jpetto committed:
Remove --profile from S3 command in favor of IAM role

pmac commented:
This looks great. Nice work 💯
